dandi/download.py

```diff
@@ -1122,19 +1123,36 @@ def _check_if_more_attempts_allowed(
             return None
         elif retry_after := exc.response.headers.get("Retry-After"):
```

Comment:
This is not about the changes in this PR. However, since are trying to improve this function, we may as well consider replace `elif` with `if` here. The last `if` clause ended with a `return` statement, so the `elif` here is about confusing.

Reply:
So can be said about the containing `elif` clause at line 1111 at which I can't place a comment.